### ISSUE TITLE
Error when `limit>500`

### ISSUE THREAD

**Issue by bschilder** [monarch-initiative/monarchr#14]
Trying to mass query all phenotypes in the KG by setting query to NULL and raising limit.

Running with limit<=500 works fine:
```R
out <- monarchr::monarch_search(query = NULL,
                                      category = "biolink:PhenotypicFeature", 
                                      limit = 500)
```
![Screenshot 2024-04-15 at 14 05 18](https://github.com/monarch-initiative/monarchr/assets/34280215/c75d7a1b-b4b4-48a0-8bff-93cb278ce700)

But after some trial and error, I found out that anything >500 causes an error:

```R
out <- monarchr::monarch_search(query = NULL,
                                      category = "biolink:PhenotypicFeature", 
                                      limit = 501)
```
```R
Error in response_content$total : 
  $ operator is invalid for atomic vectors
```

Is this something caused by monarchR or the KG API itself?

Thanks!
Brian

## Session info
```
R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.4.1

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/London
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] monarchr_0.2.1

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       httr_1.4.7        cli_3.6.2         knitr_1.45        rlang_1.1.3       xfun_0.43         purrr_1.0.2      
 [8] generics_0.1.3    assertthat_0.2.1  jsonlite_1.8.8    glue_1.7.0        htmltools_0.5.8.1 fansi_1.0.6       rmarkdown_2.26   
[15] neo2R_2.4.2       evaluate_0.23     tibble_3.2.1      base64enc_0.1-3   fastmap_1.1.1     yaml_2.3.8        lifecycle_1.0.4  
[22] compiler_4.3.1    dplyr_1.1.4       igraph_2.0.3      pkgconfig_2.0.3   tidyr_1.3.1       rstudioapi_0.16.0 digest_0.6.35    
[29] R6_2.5.1          tidyselect_1.2.1  utf8_1.2.4        curl_5.2.1        pillar_1.9.0      magrittr_2.0.3    tidygraph_1.3.1  
[36] tools_4.3.1    
```

**Comment by oneilsh** [comment]
This appears to be caused by the Monarch API's search endpoint, which returns a 500 when the limit > 500 (tried over at https://api-v3.monarchinitiative.org/v3/docs#/search/search_v3_api_search_get).

I'm not sure if it's feasible to get that bumped but we should at least handle it gracefully in the package :) 

Regarding bulk "grab all nodes of category X" kinds of queries, that seems like something we can better handle with the graph engine/neo4j than the keyword search (I've also not tried NULL for the query on that one, hadn't thought of that use case!)

**Comment by bschilder** [comment]
> This appears to be caused by the Monarch API's search endpoint, which returns a 500 when the limit > 500 (tried over at https://api-v3.monarchinitiative.org/v3/docs#/search/search_v3_api_search_get).
> 
> I'm not sure if it's feasible to get that bumped but we should at least handle it gracefully in the package :)
> 


Makes sense, I figured it was something like that.

> Regarding bulk "grab all nodes of category X" kinds of queries, that seems like something we can better handle with the graph engine/neo4j than the keyword search (I've also not tried NULL for the query on that one, hadn't thought of that use case!)

Totally agree, using the KG for bulk queries would be super helpful here.

**Comment by oneilsh** [comment]
Closing, these are addressed in the new 1.0 release; `fetch_nodes()` now handles fetching by query ID and by bulk node properties, whereas `monarch_search()` uses the monarch search API, and properly handles limit > 500 case.
